core/codec/leb128/leb128.hpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_CORE_CODECS_LEB128_HPP
+#define FILECOIN_CORE_CODECS_LEB128_HPP
+
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+#include "codec/leb128/leb128_decode_stream.hpp"
+#include "codec/leb128/leb128_encode_stream.hpp"
+
+namespace codec::leb128 {
+
+  /**
+   * @brief LEB128 encoding
+   * @tparam Primitive type to be encoded
+   * @param arg data to be encoded
+   * @return encoded data
+   */
+  template <typename T>
+  std::vector<uint8_t> encode(T &arg) {
+    EncodeStream stream{};
+    std::vector<uint8_t> data;
+    stream << std::forward<T>(arg);
+    data << stream;
+    return data;
+  }
+
+  /**
+   * @brief LEB128 encoding of variadic arguments
+   * @tparam Primitive types to be encoded
+   * @param args data to be encoded
+   * @return encoded data
+   */
+  template <typename... Args>
+  std::vector<uint8_t> encode(Args &&... args) {
+    EncodeStream stream{};
+    std::vector<uint8_t> data;
+    std::tuple<Args...> vars{args...};
+    std::apply([&stream](auto &&... var) { ((stream << var), ...); }, vars);
+    data << stream;
+    return data;
+  }
+
+  /**
+   * @brief LEB128 decoding
+   * @tparam T primitive type that is decoded from provided byte vector
+   * @param data byte vector with encoded data
+   * @return decoded T
+   */
+  template <class T>
+  T decode(const std::vector<uint8_t> &data) {
```

Comment:
use outcome here. https://github.com/soramitsu/kagome/blob/master/docs/result.md

Reply:
Fixed to 
template <class T>
fc::outcome::result<T> decode(const std::vector<uint8_t> &data);